test/e2e/common.go

```diff
@@ -507,4 +517,77 @@ func DumpResources(ctx context.Context, clusterProxy framework.ClusterProxy, nam
 		LogPath:   filepath.Join(artifactFolder, "clusters", clusterProxy.GetName(), "resources"),
 	})
 	dumpCRDS(ctx, clusterProxy.GetClient(), filepath.Join(artifactFolder, "crd"))
+	DumpIronicNodes(ctx, artifactFolder)
+}
+
+func DumpIronicNodes(ctx context.Context, artifactFolder string) {
+	ironicProvisioningIP, exists := os.LookupEnv("IRONIC_PROVISIONING_IP")
+	if !exists {
+		Logf("Running fixture tests, skip fetching ironic nodes.")
+		return
+	}
+	ironicURL := fmt.Sprintf("https://%s/v1/nodes", net.JoinHostPort(ironicProvisioningIP, "6385"))
+	username := os.Getenv("IRONIC_USERNAME")
+	password := os.Getenv("IRONIC_PASSWORD")
```

Comment:
It is not so nice to read environment variables so "late"/"deep" in the code. Usually things like this would be handled immediately at the start of execution of a program, to initialize data structures and configuration (just like we do in [e2e_suite_test.go](https://github.com/metal3-io/baremetal-operator/blob/main/test/e2e/e2e_suite_test.go). Then this data would be explicitly passed through as arguments so it is clear what input each function has to work with.
Now that this function reads it from environment variables, it looks a bit like magic. It doesn't take any credentials or even an endpoint as input but still knows how to get the ironic nodes.

I would want it to work more like [PerformSSHBootCheck](https://github.com/metal3-io/baremetal-operator/blob/1c72ffe4ae8b05067a85b40ed925478a657b31a1/test/e2e/common.go#L276). It takes all information it needs as arguments and it uses the e2e config for variables. That way the user gets unified way of configuring things. They can decide if they want environment variables or config files or both.

Could you add these variables to the e2e config and make this function access them that way?

Reply:
Thanks. I've now implemented it that way. Could you check and see if the code looks like what you have in mind?